servers/services/src/main/java/org/projectnessie/services/impl/NamespaceApiImpl.java

```diff
@@ -114,9 +114,7 @@ public void deleteNamespace(NamespaceParams params)
             try (Stream<KeyEntry<Type>> keys = getStore().getKeys(branch)) {
               if (keys.anyMatch(
                   k ->
-                      Namespace.of(k.getKey().getElements())
-                              .name()
-                              .startsWith(params.getNamespace().name())
+                      Namespace.of(k.getKey().getElements()).startsWith(params.getNamespace())
```

Comment:
Currently deleting a `Namespace.of("a", "b")` will delete `Key.of("a", "bc", "d")`... This does not look correct to me.

Reply:
in hindsight I agree with your observation. I think we actually need to do equality comparisons for each individual element